gui/common/freenasldap.py

```diff
@@ -716,22 +716,32 @@ def get_kerberos_ticket(self):
 
         elif self.krb_realm and self.binddn and self.bindpw:
             user = self.get_user_by_DN(self.binddn)
-            uid = user[1]['uid'][0]
+ 
+            try: 
+                uid = user[1]['uid'][0].decode('utf-8')
+            except:
+                uid = user[1]['uid'][0]
+
+            try: 
+                bindpw = self.bindpw.encode('utf-8')
+            except: 
+                bindpw = self.bindpw 
 
             krb_principal = self.get_kerberos_principal_from_cache()
             principal = "%s@%s" % (uid, self.krb_realm)
 
             if krb_principal and krb_principal.upper() == principal.upper():
                 return True
 
-            f = tempfile.NamedTemporaryFile(mode='w+', dir="/tmp")
-            os.chmod(f.name, 0o600)
-            f.write(self.bindpw)
+            (fd, fname) = tempfile.mkstemp(dir="/tmp", text=True)
+            os.write(fd, bindpw)
+            os.fchmod(fd, 0o777)
```

Comment:
Do we really need to make file with password 0x777?

Reply:
No. Clearly this is wrong.